srcdeps-core/src/main/java/org/srcdeps/core/config/scalar/CharStreamSource.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2015-2017 Maven Source Dependencies
+ * Plugin contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.srcdeps.core.config.scalar;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.io.StringReader;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * An URI-like combination of {@link Scheme} and some scheme specific {@code resource} identifier.
+ * <ul>
+ * <li>For {@code classpath} scheme, the {@code resource} is a path that will be resolved via
+ * {@code getClass().getResource(resource)}. Example:
+ * {@code classpath:/gradle/settings/srcdeps-model-transformer.gradle}</li>
+ *
+ * <li>For {@code file} scheme, the {@code resource} is a file system path relative to the outer project's root
+ * directory. Example: {@code file:src/build/srcdeps/my-script.gradle}</li>
+ *
+ * <li>For {@code literal} scheme, the {@code resource} is a string literal to use verbatim as the resource content.
+ * {@code literal: println "srcdeps rocks!"}
+ * <ul>
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class CharStreamSource {
+
+    public enum Scheme {
+        classpath {
```

Comment:
Shouldn't Enums follow the conventions of constants? So, `CLASSPATH`, instead of `classpath`?

Reply:
No, as long as the given enum's entry names are supposed to be used verbatim when parsing/serializing 8-)